tutorials/shaders/screen-reading_shaders.rst

```diff
@@ -102,56 +104,94 @@ With correct back-buffer copying, the two spheres blend correctly:
-   automatic copying of the entire screen only happens if ``SCREEN_TEXTURE`` is
-   used in a node for the first time and no BackBufferCopy node (not
-   disabled) was found before in tree-order.
--  BackBufferCopy can copy either the entire screen or a region. If set
-   to only a region (not the whole screen) and your shader uses pixels
-   not in the region copied, the result of that read is undefined
-   (most likely garbage from previous frames). In other words, it's
-   possible to use BackBufferCopy to copy back a region of the screen
-   and then use ``SCREEN_TEXTURE`` on a different region. Avoid this behavior!
-
-
-DEPTH_TEXTURE
+-  If a BackBufferCopy node was processed before the situation in the point
+   above (even if ``hint_screen_texture`` was not used), the behavior described
+   in the point above does not happen. In other words, automatic copying of the
+   entire screen only happens if ``hint_screen_texture`` is used in a node for
+   the first time and no BackBufferCopy node (not disabled) was found before in
+   tree-order.
+-  BackBufferCopy can copy either the entire screen or a region. If set to only
+   a region (not the whole screen) and your shader uses pixels not in the region
+   copied, the result of that read is undefined (most likely garbage from
+   previous frames). In other words, it's possible to use BackBufferCopy to copy
+   back a region of the screen and then read the screen texture from a different
+   region. Avoid this behavior!
+
+
+Depth texture
 ~~~~~~~~~~~~~
 
 For 3D shaders, it's also possible to access the screen depth buffer. For this,
-the ``DEPTH_TEXTURE`` built-in is used. This texture is not linear; it must be
-converted via the inverse projection matrix.
+the ``hint_depth_texture`` hint is used. This texture is not linear; it must be
+converted using the inverse projection matrix.
 
 The following code retrieves the 3D position below the pixel being drawn:
 
 .. code-block:: glsl
 
+    uniform sampler2D depth_texture : hint_depth_texture, repeat_disabled, filter_nearest;
+
     void fragment() {
-        float depth = textureLod(DEPTH_TEXTURE, SCREEN_UV, 0.0).r;
+        float depth = textureLod(depth_texture, SCREEN_UV, 0.0).r;
         vec4 upos = INV_PROJECTION_MATRIX * vec4(SCREEN_UV * 2.0 - 1.0, depth, 1.0);
         vec3 pixel_position = upos.xyz / upos.w;
     }
+
+Normal-roughness texture
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Similarly, the normal-roughness texture can be used to read the normals and
+roughness of objects rendered in the depth prepass. The normal is stored in the
+``.xyz`` channels while the roughness is stored in the ``.w`` channel.
+
+.. code-block:: glsl
+
+    uniform sampler2D normal_roughness_texture : hint_normal_roughness_texture, repeat_disabled, filter_nearest;
+
+    void fragment() {
+        float screen_roughness = texture(normal_roughness_texture, SCREEN_UV).w;
+        vec3 screen_normal = texture(normal_roughness_texture, SCREEN_UV).xyz;
```

Comment:
Hi, thanks a lot for your work on this!
I think it would be nice to mention somewhere how exactly the normals are represented here (xyz values are mapped to a range of 0-1) and maybe also add a line that restores the actual normals.
```suggestion
        vec3 screen_normal = texture(normal_roughness_texture, SCREEN_UV).xyz;
        screen_normal = screen_normal * 2.0 - 1.0;
```

Reply:
Thanks a great idea! I made the change